### PR TITLE
fix: always ensure the base command is set to "sam"

### PR DESCRIPTION
**Pull request by TheSriram**
- irrespective of the module name that invokes the CLI, the command that
  is seen on the help/description text, should always say "sam"

Example

```
(aws_sam_cli_WS) srirammv@localrainbow:20:20:34:~/aws_sam_cli_WS/aws-sam-cli$python -m samcli
Usage: sam [OPTIONS] COMMAND [ARGS]...

  AWS Serverless Application Model (SAM) CLI

  The AWS Serverless Application Model extends AWS CloudFormation to provide
  a simplified way of defining the Amazon API Gateway APIs, AWS Lambda
  functions, and Amazon DynamoDB tables needed by your serverless
  application. You can find more in-depth guide about the SAM specification
  here: https://github.com/awslabs/serverless-application-model.

Options:
  --debug    Turn on debug logging to print debug message generated by SAM
             CLI.
  --version  Show the version and exit.
  --info
  --help     Show this message and exit.

Commands:
  publish   Publish a packaged AWS SAM template to the AWS Serverless
            Application Repository.
  package   Package an AWS SAM application. This is an alias for 'aws
            cloudformation package'.
  init      Initialize a serverless application with a...
  build     Build your Lambda function code
  validate  Validate an AWS SAM template.
  local     Run your Serverless application locally for...
  logs      Fetch logs for a function
  deploy    Deploy an AWS SAM application. This is an alias for 'aws
            cloudformation deploy'.
```

we now see `sam [OPTIONS] COMMAND [ARGS]...` instead of `__main__.py [OPTIONS] COMMAND [ARGS]...`
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samcli/cli/command.py [hunk 1]
```diff
@@ -110,3 +111,28 @@ def get_command(self, ctx, cmd_name):
             return
 
         return mod.cli
+
+    # NOTE(TheSriram): info_name is always set to "sam". This way when the CLI is invoked as a module,
+    # the help text that is generated still says "sam" instead of "__main__".
+    def make_context(self, info_name, args, parent=None, **extra):
```
Comment (TheSriram) [inline samcli/cli/command.py]: Original make_context function: https://github.com/pallets/click/blob/38a2712d596a5df9288c1817748b71197d8c7c57/click/core.py#L621

Comment (TheSriram) [reply samcli/cli/command.py]: https://github.com/pallets/click/issues/82  is a better way to do it.

Comment (jfuss) [reply samcli/cli/command.py]: That is super clean. Nice find!

**Comment by TheSriram** [comment]
restarting build, due to transient failure.

**Pull request merged**